### ISSUE TITLE
call2 panics if not all input was consumed

### ISSUE THREAD

**Issue by Tamschi** [Tamschi/call2-for-syn#1]
As the title says, `call2` currently panics if `parser` does not consume all of the input. This is inconvenient.

The programming error is this line: `parse2::<Dummy>(input).unwrap();`

Instead of unwrapping here, the function should return a `Result<T, Error<T>>` where

```rust
#[non_exhaustive]
enum Error<T> {
  PartiallyParsed {
    parsed: T,
    parse2_error: syn::Error,
  },
}
```

.

There should also be an extension trait that provides `Result<T, Error<T>>::accept_partially_parsed(self) -> T`.

**Comment by Tamschi** [comment]
Thinking about it, a more straightforward error type and an alternative `call2_allow_incomplete` that just returns `T` are probably better.

**Comment by Tamschi** [comment]
Due to #4, the location of the panic has shifted to `Parser::parse2(…).unwrap()`.

**Comment by Tamschi** [comment]
Fixed by fe0e98c.
